functions/src/functions/types.ts

```diff
@@ -0,0 +1,7 @@
+export interface Result<T> {
```

Comment:
I think we could make it smarter with union. This way, there is no need for double-checking of `data`/`error` existance. 

Take a look at [this TS REPL](https://www.typescriptlang.org/play/?ssl=12&ssc=1&pln=4&pc=1#code/KYDwDg9gTgLgBDAnmYcBKwDOBXANjAHgBUA+OAXjgG84ATAQxnoC44iAaOYKKaAflbYAdrWAAzAJZDgtOAF84AH2p1GLOMNGTptTt15RWVAMYRRrTDChSA5pwC2WTPRvALV2-LkBYAFB-TIUs4TAhHAAl6EVxuCjgACigsSCC3dCw8QktrIRsSAEoKMio-OAkxBKTMFMxgADp9aEKS3zg2uAB6DrgqmvrGqDgAC25UCUw4CAAjACtgY3gAdwkYIZVTcxCPXIcnFzTszx9WhWBcWupS9q6e5IhUuoYmBGRUEaSyifpcJPpaRDgQnoPAgixkCAgWxyNiubV691qjzUdWMQ2BAEEYPEAKz5UrHY5+IA)

Reply:
sounds good to me - thank you for the suggestion!